drivers/platform/mbed/mbed_timer.cpp

```diff
@@ -0,0 +1,191 @@
+ * @param desc[in, out] - The Timer descriptor.
+ * @param init_param[in] - Structure that contains Timer Initialization Parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t timer_init(struct timer_desc **desc,
+		   struct timer_init_param *param)
+{
+	struct timer_desc *timer_desc; 		// Pointer to timer descriptor
+	struct mbed_timer_desc *mbed_timer_desc;	// Pointer to mbed timer descriptor
+	mbed::Timer *timer;					// Pointer to new Timer instance
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	timer_desc = (struct timer_desc *)calloc(1, sizeof(*timer_desc));
+	if (!timer_desc)
+		return -ENOMEM;
+
+	mbed_timer_desc = (struct mbed_timer_desc *)calloc(1, sizeof(*mbed_timer_desc));
+	if (!mbed_timer_desc)
+		goto err_mbed_timer_desc;
+
+	timer = new Timer();
+	if (!timer)
+		goto err_timer;
+
+	mbed_timer_desc->timer = timer;
+	timer_desc->extra = mbed_timer_desc;
+
+	*desc = timer_desc;
+	return SUCCESS;
+
+err_timer:
+	free(mbed_timer_desc);
+err_mbed_timer_desc:
+	free(timer_desc);
+
+	return -ENOMEM;
+}
+
+/**
+* @brief Start the Timer.
+* @param desc[in] - The Timer Descriptor.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t timer_start(struct timer_desc *desc)
+{
+	mbed::Timer *timer;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	timer = (Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer);
+	if (!timer)
+		return -ENOMEM;
+
+	timer->start();
+
+	return SUCCESS;
+}
+
+/**
+* @brief Stop the Timer.
+* @param desc[in] - The Timer Descriptor.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t timer_stop(struct timer_desc *desc)
+{
+	mbed::Timer *timer;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	timer = (Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer);
+	if (!timer)
+		return -ENOMEM;
+
+	timer->stop();
+
+	return SUCCESS;
+}
+
+/**
+* @brief Release all the resources allocated by Timer.
+* @param desc[in] - The Timer Descriptor.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t timer_remove(struct timer_desc *desc)
+{
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	/* Free the timer object */
+	if ((Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer))
+		delete((Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer));
+
+	/* Free the Timer descriptors */
+	free(desc->extra);
+	free(desc);
+
+	return SUCCESS;
+}
+
+/**
+* @brief Get the elapsed time in nanoseconds.
+* @param desc[in] - The Timer descriptor.
+* @param elapsed_time[in, out] - Pointer where the elapsed time value is stored.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t get_elapsed_time_in_nsec(struct timer_desc *desc,
+				 uint64_t *elapsed_time)
+{
+	mbed::Timer *timer;
+
+	if (!desc || !desc->extra || !elapsed_time)
+		return -EINVAL;
+
+	timer = (Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer);
```

Comment:
ditto as in https://github.com/analogdevicesinc/no-OS/pull/1243/files#r802329865